node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1134,10 +1383,16 @@ async fn fetch_to_genesis(trusted_block: &Block, ctx: &ChainSyncContext<'_>) ->
 const MAX_HEADERS_BATCH_SIZE: u64 = 1024;
 
 // Fetches headers starting from `trusted_block` till the Genesis.
-async fn fetch_headers_till_genesis(
+async fn fetch_headers_till_genesis<REv>(
     trusted_block: &Block,
-    ctx: &ChainSyncContext<'_>,
-) -> Result<(), Error> {
+    ctx: &ChainSyncContext<'_, REv>,
```

Comment:
In the future we could make all of those functions methods of `ChainSyncContext`.

Reply:
Well, I'd quite like to refactor this into a more "normal" component eventually.  The `ChainSyncContext` is pretty much what the component should be imo.  But we can discuss any large change like that before it gets ticketed.